theano/tensor/tests/test_elemwise.py

```diff
@@ -826,6 +826,7 @@ def test_reduce_default_acc_dtype(self):
                     uint8='uint64',
                     uint16='uint64',
                     uint32='uint64',
+                    float16='float32',
```

Comment:
This is in the tests.  And float16 does uses float32 by default for accumulation.  If we want to change that I'm ok with it, but I would make that change in another PR.